qiskit_optimization/applications/ising/base_application.py

```diff
@@ -0,0 +1,25 @@
+import copy
+from abc import ABC, abstractmethod
+
+
+class BaseApplication(ABC):
+    """
+    An abstract class for optimization problems
+    """
+
+    @abstractmethod
+    def _build_quadratic_program(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def interpret():
+        raise NotImplementedError
+
+    def is_feasible(self, x):
+        return self._build_quadratic_program().is_feasible(x)
+
+    def evaluate(self, x):
+        return self._build_quadratic_program().objective.evaluate(x)
+
+    def to_quadratic_program(self):
+        return copy.deepcopy(self._qp)
```

Comment:
Why use `self._qp` here but not in `evaluate` and `is_feasible` ? If we store it, we need to make sure we delete it or re-build it in case anything changes in an application object.

Reply:
If we want to have a BaseApplication class here would it make sense to somehow separate the problem part of the application, that perhaps can be treated more generically, from the aspect of the quadratic_program that may not be the only path in the future to solving the problem?

In respect of this class on self-._qp it refers to a variable that is not part of this class. Maybe there should be some private method to set it and/or pass it up to a constructor here to store it. Maybe it supposed to be stored/cached via the build_quadratic_program, in some manner.

Reply:
As the class is now being used in another domain, I know naming may not be sorted yet, but perhaps when it is something like OptimizationApplication may be a more apt base class name that when looking at a FinanceApp, that is derived, its more obvious than a generically named BaseApplication